migrations/2022-04-04-183652_update_community_aggregates_on_soft_delete/up.sql

```diff
@@ -0,0 +1,64 @@
+drop trigger if exists community_aggregates_post_count on post;
+drop trigger if exists community_aggregates_comment_count on comment;
+
+create or replace function community_aggregates_post_count()
+    returns trigger language plpgsql
+as $$
+begin
+  IF (TG_OP = 'INSERT') THEN
+update community_aggregates
+set posts = posts + 1 where community_id = NEW.community_id;
+ELSIF (TG_OP = 'DELETE'  OR (TG_OP = 'UPDATE' AND OLD.deleted = 'f' AND NEW.deleted = 't')) THEN
+update community_aggregates
+set posts = posts - 1 where community_id = OLD.community_id;
+
+-- Update the counts if the post got deleted
+update community_aggregates ca
+set posts = coalesce(cd.posts, 0),
+    comments = coalesce(cd.comments, 0)
+    from ( 
+      select 
+      c.id,
+      count(distinct p.id) as posts,
+      count(distinct ct.id) as comments
+      from community c
+      left join post p on c.id = p.community_id and p.deleted = 'f'
+      left join comment ct on p.id = ct.post_id and ct.deleted = 'f'
+      group by c.id
+    ) cd
+where ca.community_id = OLD.community_id;
+END IF;
+return null;
+end $$;
+
+create trigger community_aggregates_post_count
```

Comment:
Did you test this? I'm guessing its going to tell you that this trigger already exists.

Reply:
Yes, it did work for me. To make sure that the triggers are replaced, I included the `drop trigger if exists` statements at the top of both files.